pkg/sql/plan2/function/function.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/errno"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/sql/errors"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+const (
+	// NullType means the type of constant value `NULL`
+	// which can meet each required type.
+	// e.g.
+	// if we input a SQL `select built_in_function(100, NULL);`
+	// it will use [types.T_int64, NullType] to search function when we were building the query plan.
+	NullType = types.T_any
```

Comment:
NULL type is definitely not ANY -- Here, the null type wants to say, this value is a const null, actually, it is not even a type.    ANY means we don't know, the type can be anything.

I am not sure here you have misnomer, or wrong semantics.

Reply:
There should be a wording error here, what he wants to express is the scalar null, but since null is not a type, every type can theoretically have null, so he uses the word any without knowing how to express it, there may be a conflict between any and any of the types, it may be better to change it to anyNULL or some other word.

Reply:
what I mean is what nnsgmsone said. I use it to represent scalar NULL.
Maybe types.T_any is inaccuracy. Use types.T_unknown is better.

It's easy to change.